### ISSUE TITLE
Unknown step type: *DiscardOpenChangesStep

### ISSUE THREAD

**Issue by cirego** [git-town/git-town#1628]
I'm not exactly sure how I got this error, but I was trying to run `git town ship`. It popped up with a merge conflict that it wanted me to resolve. I then tried `git town continue` and got the following error message:

```
[main] git merge --abort                                   
fatal: There is no merge to abort (MERGE_HEAD missing).                                                                
                                                           
Error: cannot run the abort steps: cannot abort current merge: exit status 128                                                                                                                                                                
                                                                                                                       
To abort, run "git-town abort".              
To continue after having resolved conflicts, run "git-town continue".  
```

I then tried `git town abort` and that's when I got this error message: `2020/08/05 13:59:53 Unknown step type: *DiscardOpenChangesStep`

I'm running `Git Town v7.4.0-dev (2020/07/29)`, built from `ca6f439a72d9ddbc60e7833c427235a1e9ed7e40`.

**Comment by kevgo** [comment]
Thanks for the bug report! The error is thrown at this line: https://github.com/git-town/git-town/blob/c54df1141a327fac987e96195ceb16df24ddaaa1/src/steps/json_step.go#L108-L110

We need to add `DiscardOpenChangesStep` to the switch statement.

Regarding reproducibility, I'm puzzled that it tries running `git merge --abort`. This is part of the `AbortMerge` method of `git.Runner`: https://github.com/git-town/git-town/blob/b70bcf3fba27e0620bcb3b771b03d4bc72ec67e2/src/git/runner.go#L31-L33

Are you sure that you didn't run `git town abort` after resolving the merge conflict?

**Comment by cirego** [comment]
Hey kevgo! Thanks for the quick response. I'm not sure why it ran `merge --abort` but I'm certain that I did not run `git town abort`.

**Comment by kevgo** [comment]
I have shipped #1631 which adds the missing `DiscardOpenChangesStep` handling. If you can repro why it ran `git merge --abort` without you running `git town abort`, please file another bug report. Thanks!

**Comment by cirego** [comment]
Thank you! Will keep my eyes open for a repro.
